src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -0,0 +1,238 @@
+}
+
+void ImpedanceTask::addToLogger(mc_rtc::Logger & logger)
+{
+  SurfaceTransformTask::addToLogger(logger);
+
+  // impedance parameters
+  logger.addLogEntry(name_ + "_impedanceM", [this]() -> const sva::ForceVecd & { return impM_; });
+  logger.addLogEntry(name_ + "_impedanceD", [this]() -> const sva::ForceVecd & { return impD_; });
+  logger.addLogEntry(name_ + "_impedanceK", [this]() -> const sva::ForceVecd & { return impK_; });
+  logger.addLogEntry(name_ + "_wrenchGain", [this]() -> const sva::MotionVecd & { return wrenchGain_; });
+
+  // compliance values
+  logger.addLogEntry(name_ + "_compliancePose", [this]() -> const sva::PTransformd & { return compPoseW_; });
+  logger.addLogEntry(name_ + "_complianceVel", [this]() -> const sva::MotionVecd & { return compVelW_; });
+  logger.addLogEntry(name_ + "_complianceAccel", [this]() -> const sva::MotionVecd & { return compAccelW_; });
+
+  // desired values
+  logger.addLogEntry(name_ + "_desiredPose", [this]() -> const sva::PTransformd & { return desiredPoseW_; });
+  logger.addLogEntry(name_ + "_desiredVel", [this]() -> const sva::MotionVecd & { return desiredVelW_; });
+  logger.addLogEntry(name_ + "_desiredAccel", [this]() -> const sva::MotionVecd & { return desiredAccelW_; });
+
+  // wrench
+  logger.addLogEntry(name_ + "_targetWrench", [this]() -> const sva::ForceVecd & { return targetWrench_; });
+  logger.addLogEntry(name_ + "_measuredWrench",
+                     [this]() -> sva::ForceVecd { return measuredWrench(); }); // should not be reference
+}
+
+void ImpedanceTask::removeFromLogger(mc_rtc::Logger & logger)
+{
+  SurfaceTransformTask::removeFromLogger(logger);
+
+  // impedance parameters
+  logger.removeLogEntry(name_ + "_impedanceM");
+  logger.removeLogEntry(name_ + "_impedanceD");
+  logger.removeLogEntry(name_ + "_impedanceK");
+  logger.removeLogEntry(name_ + "_wrenchGain");
+
+  // compliance values
+  logger.removeLogEntry(name_ + "_compliancePose");
+  logger.removeLogEntry(name_ + "_complianceVel");
+  logger.removeLogEntry(name_ + "_complianceAccel");
+
+  // desired values
+  logger.removeLogEntry(name_ + "_desiredPose");
+  logger.removeLogEntry(name_ + "_desiredVel");
+  logger.removeLogEntry(name_ + "_desiredAccel");
+
+  // wrench
+  logger.removeLogEntry(name_ + "_targetWrench");
+  logger.removeLogEntry(name_ + "_measuredWrench");
+}
+
+void ImpedanceTask::addToGUI(mc_rtc::gui::StateBuilder & gui)
+{
+  // Don't add SurfaceTransformTask because the target of SurfaceTransformTask should not be set by user
+  TrajectoryTaskGeneric<tasks::qp::SurfaceTransformTask>::addToGUI(gui);
+
+  gui.addElement({"Tasks", name_},
+                 // pose
+                 mc_rtc::gui::Transform("desiredPose", [this]() { return this->desiredPose(); },
+                                        [this](const sva::PTransformd & pos) { this->desiredPose(pos); }),
+                 mc_rtc::gui::Transform("compliancePose", [this]() { return this->compliancePose(); }),
```

Comment:
```suggestion
                 mc_rtc::gui::Transform("compliancePose", [this]() -> const sva::PTransformd & { return this->compliancePose(); }),
```

Reply:
Because of the change after your review (sorry for that), this is no longer reference: https://github.com/jrl-umi3218/mc_rtc/pull/100/commits/40265e6d8c428f77c3924a09ab3955f37c51e8f7#diff-db3809a3a52f623c53b2c5836c7546b76608fe22dd5e5e097bfc2c1e306e551dR215 .